frontend/src/hooks/useConditionallyHiddenColumns.ts

```diff
@@ -7,19 +7,27 @@ interface IConditionallyHiddenColumns {
 
 export const useConditionallyHiddenColumns = (
     conditionallyHiddenColumns: IConditionallyHiddenColumns[],
-    setHiddenColumns: (param: string[]) => void,
+    setHiddenColumns: (
+        columns: string[] | ((columns: string[]) => string[])
+    ) => void,
     columnsDefinition: unknown[]
 ) => {
     useEffect(() => {
-        const hiddenColumnsSet = new Set<string>();
-
-        conditionallyHiddenColumns
+        const columnsToHide = conditionallyHiddenColumns
             .filter(({ condition }) => condition)
-            .forEach(({ columns }) =>
-                columns.forEach(column => hiddenColumnsSet.add(column))
-            );
+            .flatMap(({ columns }) => columns);
+
+        const columnsToShow = conditionallyHiddenColumns
+            .flatMap(({ columns }) => columns)
+            .filter(column => !columnsToHide.includes(column));
 
-        setHiddenColumns([...hiddenColumnsSet]);
+        setHiddenColumns(columns => [
+            ...new Set(
+                [...columns, ...columnsToHide].filter(
+                    column => !columnsToShow.includes(column)
+                )
+            ),
+        ]);
```

Comment:
Refactored so we take into account the previous hidden columns state.

This is especially useful in cases like `ProjectFeatureToggles` that includes user-toggle-able columns: https://github.com/Unleash/unleash/pull/2695/files#diff-a1c10fdcd539d988f74d4c3742cc6b8844152999a6c7e509ed8b1765c604524fL433-R443

This way we can simply define our visibility conditions when using the hook and it will take care of it seamlessly.